commcare_connect/opportunity/tables.py

```diff
@@ -947,3 +947,290 @@ def render_icons(self, record):
+        total_seconds = int(value.total_seconds() if isinstance(value, timedelta) else 0)
+        return get_duration_min(total_seconds)
+
+
+class WorkerStatusTable(tables.Table):
+    index = IndexColumn()
+    user = UserInfoColumn()
+    suspended = SuspendedIndicatorColumn()
+    invited_date = tables.DateColumn(format="d-M-Y")
+    last_active = tables.DateColumn(format="d-M-Y")
+    started_learn = tables.DateColumn(format="d-M-Y", verbose_name="Started Learn", accessor="date_learn_started")
+    completed_learn = tables.DateColumn(format="d-M-Y")
+    days_to_complete_learn = DurationColumn()
+    first_delivery = tables.Column()
+    days_to_start_delivery = DurationColumn()
+
+    def __init__(self, *args, **kwargs):
+        self.use_view_url = True
+        super().__init__(*args, **kwargs)
+
+    def render_lastActive(self, value):
+        return format_html(
+            """
+            <div class="flex flex-col items-start">
+                <p class="text-sm text-slate-900 ">{}</p>
+            </div>
+            """,
+            value,
+        )
+
+
+class WorkerPaymentsTable(tables.Table):
+    index = IndexColumn()
+    user = UserInfoColumn()
+    suspended = SuspendedIndicatorColumn()
+    last_active = tables.Column()
+    payment_accrued = tables.Column(verbose_name="Accrued")
+    total_paid = tables.Column()
+    last_paid = tables.DateColumn(format="d-M-Y")
+    total_confirmed_paid = tables.Column(verbose_name="Confirm")
+
+    def __init__(self, *args, **kwargs):
+        self.use_view_url = True
+        super().__init__(*args, **kwargs)
+
+    class Meta:
+        model = OpportunityAccess
+        fields = ("user", "suspended", "payment_accrued", "total_paid", "total_confirmed_paid")
+        sequence = (
+            "index",
+            "user",
+            "suspended",
+            "last_active",
+            "payment_accrued",
+            "total_paid",
+            "last_paid",
+            "total_confirmed_paid",
+        )
+
+
+class WorkerLearnTable(OrgContextTable):
+    index = IndexColumn()
+    user = UserInfoColumn()
+    suspended = SuspendedIndicatorColumn()
+    last_active = tables.DateColumn(format="d-M-Y")
+    started_learning = tables.DateColumn(
+        format="d-M-Y", accessor="date_learn_started", verbose_name="Started Learning"
+    )
+    modules_completed = tables.TemplateColumn(
+        accessor="modules_completed_percentage",
+        template_code="""
+                            {% include "tailwind/components/progressbar/simple-progressbar.html" with text=flag progress=value|default:0 %}
+                        """,
+    )
+    completed_learning = tables.DateColumn(
+        format="d-M-Y", accessor="completed_learn", verbose_name="Completed Learning"
+    )
+    assessment = tables.Column(accessor="passed_assessment")
+    attempts = tables.Column(accessor="assesment_count")
+    learning_hours = DurationColumn()
+    action = tables.TemplateColumn(
```

Comment:
Figure out which of  `action` attr or `render_action` is needed (and the right column-class).

Reply:
Cleaned up the redundant TemplateCode in here